1678. Goal Parser Interpretation

```diff
@@ -0,0 +1,19 @@
+class Solution(object):
+    def interpret(self, command):
+        """
+        :type command: str
+        :rtype: str
+        """
+        res = ""
+        i = 0
+        while i < len(command):
+            if command[i] == "G":
+                res += "G"
+                i += 1
+            elif command[i] == "(" and command[i+1] == ")":
+                res += "o"
+                i += 2
+            else:
+                res += "al"
+                i += 4
```

Comment:
won't this also cause an out of bounds error? or are there some special rules in the leetcode problem here that prevent this

Reply:
leetcode problem has special rules I believe